### PR TITLE
[docs/core] Batch training on Core example (SSML)

### PR DESCRIPTION
**Pull request by Yard1**
Signed-off-by: Antoni Baum <antoni.baum@protonmail.com>

## Why are these changes needed?

This PR adds a notebook showcasing how to conduct batch training & tuning on Ray Core. This is a common SSML use case we see with users. While this is possible using higher-level APIs, a self contained example like this is helpful for:
* Use cases not covered by our higher level APIs
* Use cases requiring higher performance than what we currently provide with higher-level APIs
* Showcasing the correct usage patterns of Ray Core primitives

A notebook solving the exact same problem but with Ray Data & AIR will be added in a followup and linked from this notebook later.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Yard1** [comment]
Currently failing in CI due to an S3 authentication error, working on it

File: doc/source/ray-core/tips-for-first-time.rst [hunk 1]
```diff
@@ -240,6 +240,8 @@ Running the above program on a 2018 MacBook Pro notebook shows:
 
 In other words, it takes almost half a millisecond to execute an empty task. This suggests that we will need to make sure a task takes at least a few milliseconds to amortize the invocation overhead. One caveat is that the per-task overhead will vary from machine to machine, and between tasks that run on the same machine versus remotely. This being said, making sure that tasks take at least a few milliseconds is a good rule of thumb when developing Ray programs.
 
+.. _tip-avoid-same-object-in-remote:
```
Comment (jjyao) [inline doc/source/ray-core/tips-for-first-time.rst:243]: This has been converted to an anti-pattern (https://docs.ray.io/en/master/ray-core/patterns/pass-large-arg-by-value.html), could you reference that instead.

**Comment by Yard1** [comment]
CI now passes.

**Review by richardliaw — CHANGES_REQUESTED** [review]
Great example! A couple things:

1. Can we add some diagrams in the beginning to show exactly what is the workload that you're about to demonstrate? Batch training isn't a very common term (afaik some people use "partitioned training" too)
2. Why use pyarrow instead of pandas? the pyarrow API is not as commonly used
3. for run(), can you bring out the dataset construction so it looks like:

```
dataset = ...
models = []
generator = task_generator(dataset, models)
run(generator)
```

otherwise it's just not apparent where the dataset is constructed.

4. instead of the long output, do you think you could just show a tqdm output?

**Comment by Yard1** [comment]
@richardliaw Applied feedback

Re. 2 - pyarrow has push-down predicates which allows us to load only the rows necessary into memory. Clarified that in the notebook
Re. 4 - we can't use tqdm as we do not know ahead of time how many tasks will be ran. I made the output shorter instead.

**Review by richardliaw — APPROVED** [review]

**Pull request merged**